authority/voting/server/state_test.go

```diff
@@ -176,68 +177,77 @@ func TestVote(t *testing.T) {
+		idKey, c, err := genMixConfig(fmt.Sprintf("node-%d", i), votingPKI, port)
 		require.NoError(err)
 		mixCfgs = append(mixCfgs, c)
+		idKeys = append(idKeys, idKey)
 		port++
+		reverseHash[idKey.pubKey.Sum256()] = idKey.pubKey
 	}
 
 	// generate a Topology section
 	topology := config.Topology{Layers: make([]config.Layer, 3)}
-	topology.Layers[0].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[0].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[1].Debug.IdentityKey.PublicKey()}}
-	topology.Layers[1].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[2].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[3].Debug.IdentityKey.PublicKey()}}
-	topology.Layers[2].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[4].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[5].Debug.IdentityKey.PublicKey()}}
+	topology.Layers[0].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[0].pubKey)},
+		config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[1].pubKey)}}
+	topology.Layers[1].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[2].pubKey)},
+		config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[3].pubKey)}}
+	topology.Layers[2].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[4].pubKey)},
+		config.Node{IdentityPublicKeyPem: pem.ToPEMString(idKeys[5].pubKey)}}
 
 	// generate a conflicting Topology
 	// generate a Topology section
-	topology2 := config.Topology{Layers: make([]config.Layer, 3)}
-	topology2.Layers[0].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[0].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[1].Debug.IdentityKey.PublicKey()}}
-	topology2.Layers[1].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[2].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[3].Debug.IdentityKey.PublicKey()}}
-	topology2.Layers[2].Nodes = []config.Node{config.Node{IdentityKey: mixCfgs[5].Debug.IdentityKey.PublicKey()},
-		config.Node{IdentityKey: mixCfgs[4].Debug.IdentityKey.PublicKey()}}
+	/*
+		topology2 := config.Topology{Layers: make([]config.Layer, 3)}
+		topology2.Layers[0].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: idKeys[0].identityPublicKeyPem},
+			config.Node{IdentityPublicKeyPem: idKeys[1].identityPublicKeyPem}}
 
+		topology2.Layers[1].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: idKeys[2].identityPublicKeyPem},
+			config.Node{IdentityPublicKeyPem: idKeys[3].identityPublicKeyPem}}
+		topology2.Layers[2].Nodes = []config.Node{config.Node{IdentityPublicKeyPem: idKeys[5].identityPublicKeyPem},
+			config.Node{IdentityPublicKeyPem: idKeys[4].identityPublicKeyPem}}
+
+	*/
 	// one auth uses the conflicting topology, so we shall expect consensus with 2/3
 	authCfgs[0].Topology = &topology
-	authCfgs[1].Topology = &topology2
+	authCfgs[1].Topology = &topology
```

Comment:
we changed this test to target full consensus instead of a limited threshold. we should remove the `conflicting topology` comment above if we want to keep that.